helpers/all.js

```diff
@@ -0,0 +1,52 @@
+var _ = require('lodash'),
+    internals = {};
+
+/**
+ * Yield block only if all arguments are valid
+ *
+ * @example
+ * {{#all items theme_settings.optionA theme_settings.optionB}} ... {{/all}}
+ */
+
+internals.implementation = function(handlebars) {
+    this.handlebars = handlebars;
+};
+
+internals.implementation.prototype.register = function(context) {
+    this.handlebars.registerHelper('all', function () {
+
+        var args = [], opts, result;
+
+        // Translate arguments to array safely
+        for (var i = 0; i < arguments.length; i++) {
+            args.push(arguments[i]);
+        }
+
+        // Take the last argument (content) out of testing array
+        opts = args.pop();
+
+        // Check if all the arguments are valid / truthy
+        result = _.all(args, function(arg) {
```

Comment:
I'll note in my original code I did something with `Array.reduce`. However, we talked about having the names of the helpers consistent with lodash. Also, other helpers are using lodash, so it's not really any extra overhead, although that might depend...

:+1: I'm open to making a change if you think it's really worth it though. Maybe a minor gain to be had from using a native method vs the lodash one.